docs/src/references/hsm-support.md

```diff
@@ -1,242 +1,228 @@
-              user-pin-initialized
-              token-initialized
-...
-```
-
-7. Create systemd files:
-
-Create the following systemd files
-
-file: /etc/systemd/system/p11-kit-server.socket
-
-```toml
-[Unit]
-Description=p11-kit server
-
-[Socket]
-Priority=6
-Backlog=5
-ListenStream=%t/p11-kit/pkcs11
-SocketMode=0600
-
-[Install]
-WantedBy=sockets.target
-```
-
-file: /etc/systemd/system/p11-kit-server.service
+## Setup guide
+<!-- split the guide into a separate page under "Operate Devices" category? -->
+<!-- also would be nice to write a test for this guide -->
 
-```toml
-[Unit]
-Description=p11-kit server
-Documentation=man:p11-kit(8)
+The following guide shows how to connect to Cumulocity using a PKCS #11 cryptographic token. Instead
+of using a dedicated hardware token, we'll create a software token using
+[SoftHSM2](https://github.com/softhsm/softHSMv2) and import currently used private key on it.
 
-Requires=p11-kit-server.socket
+:::note
 
-[Service]
-Type=simple
-StandardError=journal
-ExecStart=/usr/bin/p11-kit server -f -u tedge -g tedge -n %t/p11-kit/pkcs11 pkcs11:model=PKCS%2315%20emulated;manufacturer=piv_II
-# Or use a more exact filter
-#ExecStart=/usr/bin/p11-kit server -f -u tedge -g tedge -n %t/p11-kit/pkcs11 pkcs11:model=PKCS%2315%20emulated;manufacturer=piv_II;serial=27bef941ab9ae36e;token=rpi5-d83addab8e9f
-Restart=on-failure
+While this guide uses SoftHSM2 to demonstrate the feature, be aware that in a real production
+setting you'll probably be using a different, hardware token. The process of setting up the token
+itself may be different for each token type, as well as require using a different PKCS #11 dynamic
+library, but in all cases, the goal is to:
 
-[Install]
-Also=p11-kit-server.socket
-WantedBy=default.target
-```
+- store the private key on the HSM
+- store the corresponding certificate on the file system
+- set `device.cert_path` to the certificate path
+- set `device.cryptoki.module_path` to the correct PKCS #11 dynamic library
+- set `device.cryptoki.pin` and `device.cryptoki.uri` accordingly to the local HSM settings
 
-8. Configure the opensc-pkcs11 module for p11-kit server and client
+:::
 
+### Part 1: Setup the cryptographic token
+
+1. Install SoftHSM2 (to create the token and key) and `p11tool` (to view the [PKCS #11 URI][p11uri]
+   of the key).
+
+    ```sh
+    sudo apt-get install -y softhsm2 gnutls-bin
+    ```
+
+    [p11uri]: https://www.rfc-editor.org/rfc/rfc7512
+
+For SoftHSM configuration, see [SoftHSM README](https://github.com/softhsm/softHSMv2?tab=readme-ov-file#configure-1).
+
+2. Add tedge and current user to `softhsm` group. Only users belonging to `softhsm` group can view
+   and manage SoftHSM tokens. After adding your own user, remember to logout and login for changes
+   to take effect. Alternatively, you can just run `softhsm2-util` and `p11tool` with `sudo`.
+
+   ```sh
+    sudo usermod -a -G softhsm tedge
+    sudo usermod -a -G softhsm $()
```

Comment:
What is this `$()` variable? Do you mean `$(id -un)`?

Reply:
fixed in 8fc4a62b3e